packages/expo-cli/src/commands/eas-build/build/action.ts

```diff
@@ -165,7 +170,34 @@ async function startBuildAsync<T extends Platform>(
     }
     if (!builder.ctx.commandCtx.skipProjectConfiguration) {
       try {
+        let isUpdatesConfigured = true;
+
+        switch (builder.ctx.platform) {
```

Comment:
I'm not really sure if this should be split /limited to current platform, my last comment was mostly question to @brentvatne , but if we would do that.
- this function should never have any if statements about platform, it defeats the purpose of creating builder abstraction
- if `isUpdatesConfigured` split into platforms then `setUpdateVersionsAsync` should also be this way, otherwise it could update version in unconfigured project

Reply:
Ok, I'll wait for @brentvatne before making any changes

Reply:
i think it's valuable to support doing actions for a single platform at a time.

let's say i'm setting up expo-updates for ios and then want to test doing a build. i'll run `expo eas:build -p ios` and then it will validate my config, update the necessary fields, and proceed. it would be frustrating as a user if i am blocked on needing to configure android in order to proceed.

in general we should consider extending this to the rest of the eas commands wherever possible. eg: `expo eas:build:init -p [ios/android]` would also be a valuable addition.

Reply:
We shouldn't also go to much in that direction, because every separate step adds a new prompt to commit changes.

Alternatively, we could extract code responsible for committing changes and just commit once, but it's a significant change outside the scope of this PR.

Reply:
ideally we should maintain the same behavior we have now when you initialize for both platforms, but just add on top of that the ability to initialize only for a specific platform if desired

Reply:
Why not moving this logic to the `ensureProjectConfiguredAsync` function?
I agree with @wkozyra95 that we should avoid such switch statements.